contributing/deprecation_policy.md

```diff
@@ -2,33 +2,123 @@
+However, the price paid in disruption and maintenance is real; deprecating without warning, too often,
+or for poor rationale is a hinderance to adoption and contribution. It must be done with care and
+process.
 
-## Get the 'thumbs up'
+Before embarking on the deprecation of any API we should collaboratively answer the following questions:
 
-Propose your deprecation on GitHub as an issue where the title's first phrase is "Deprecation [Component Name]". Much like a feature proposal, the contributors have a responsibility to carefully judge the consequences of breaking changes against possible gains. 
-
-We should collaboratively answer the following questions:
-
-1. Should this code be deprecated? Is it really worth the time and effort for everyone?
+1. Should this code be deprecated? Is the effort required to deprecate this component worth the benefit
+   of removing the old behavior?
 1. Why is the code being deprecated? Why should clients switch?
 1. What are clients supposed to replace the code with?
 1. What is a good deprecation timeline?
-    1. Small changes can be done in a month.
-    1. Larger changes can be done in a quarter/three months.
-    1. Adjust up or down based on the size/complexity of the change.
-1. Who’s on point? It’s good to have a single person to keep track of things.
+1. Who is responsible for the deprecation plan?
+
+## Our deprecation process (Non-Googlers)
+
+If you would like to propose that an API be deprecated please
+[file a bug](https://github.com/material-components/material-components-ios/issues/new/choose)
+explaining which API you'd like deprecated. A Googler will take the issue from there by following
+the deprecation process below.
+
+## Our deprecation process (Googlers)
+
+1. Measure internal usage of the API.
+2. Create a GitHub issue.
+3. Write a migration guide.
+4. Annotate the API as "ToBeDeprecated".
+5. Determine a schedule for the API deletion.
+6. Announce that the API will be deprecated
+7. After the deprecation date, deprecate the API.
+8. After the deletion date, re-evaluate usage of the API.
+9. Delete the API and migration guide.
+
+### Step 1: Measure internal usage of the API
+
+Read [go/mdc-ios-measuring-api-usage](http://go/mdc-ios-measuring-api-usage) to learn how to measure
+API usage internally.
+
+If there is zero API usage beyond our own library: skip to step 7 (Deprecate the API).
+Otherwise, proceed to step 2.
+
+### Step 2: Create a GitHub issue
+
+Create a GitHub issue with the following template:
+
+```
+- [x] Measure internal usage of the API.
+- [x] Create a GitHub issue.
+- [ ] Write a migration guide.
+- [ ] Annotate the API as "ToBeDeprecated".
+- [ ] Determine a schedule for the API deletion.
+- [ ] Announce that the API will be deprecated
+- [ ] After the deprecation date, deprecate the API.
```

Comment:
Consider:
After the deprecation date, annotate the API as deprecated

Reply:
Done.